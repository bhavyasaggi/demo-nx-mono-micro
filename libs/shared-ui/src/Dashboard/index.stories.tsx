import type { Meta, StoryObj } from '@storybook/react';
import { within } from '@storybook/testing-library';
import { expect } from '@storybook/jest';

import Dashboard from './index';

const meta: Meta<typeof Dashboard> = {
  component: Dashboard,
  title: 'Dashboard',
};
export default meta;

type Story = StoryObj<typeof Dashboard>;

export const DashboardPrimary = {
  args: {
    firstName: 'John',
    lastName: 'Doe',
    image: 'https://robohash.org/john.png',
    email: 'john@email.com',
    gender: 'male',
  },
};

export const DashboardStory: Story = {
  args: {
    firstName: 'John',
    lastName: 'Doe',
    image: 'https://robohash.org/john.png',
    email: 'john@email.com',
    gender: 'male',
  },
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    expect(canvas.getByText(/Welcome, John!/gi)).toBeTruthy();
  },
};
