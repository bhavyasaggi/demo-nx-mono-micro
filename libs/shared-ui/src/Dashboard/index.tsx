import {
  Avatar,
  Flex,
  Button,
  Card,
  CardFooter,
  CardBody,
  CardHeader,
  Heading,
  SimpleGrid,
  Table,
  Text,
  Tbody,
  Tr,
  Td,
  TableContainer,
} from '@chakra-ui/react';

import styles from './index.module.scss';

/* eslint-disable-next-line */
export interface DashboardProps {
  id: number;
  firstName: string;
  lastName: string;
  image: string;
  email: string;
  gender: string;
  onLogout?: () => void;
}

export function Dashboard({
  firstName,
  lastName,
  image,
  email,
  gender,
  onLogout,
}: DashboardProps) {
  const name = `${firstName} ${lastName}`;
  return (
    <Card className={`${styles['container']}`}>
      <CardHeader>
        <Heading>Welcome, {firstName}!</Heading>
      </CardHeader>
      <CardBody as={SimpleGrid} columns={2}>
        <TableContainer>
          <Table size="sm">
            <Tbody>
              <Tr>
                <Td>
                  <Text as="em">Name</Text>
                </Td>
                <Td>{name}</Td>
              </Tr>
              <Tr>
                <Td>
                  <Text as="em">Gender</Text>
                </Td>
                <Td>{gender}</Td>
              </Tr>
              <Tr>
                <Td>
                  <Text as="em">Email</Text>
                </Td>
                <Td>{email}</Td>
              </Tr>
            </Tbody>
          </Table>
        </TableContainer>
        <Flex alignItems="center" justifyContent="center">
          <Avatar
            name={name}
            src={image}
            bg="cyan"
            borderColor="teal"
            showBorder
            size="xl"
          />
        </Flex>
      </CardBody>
      <CardFooter>
        {onLogout ? (
          <Button colorScheme="red" w="100%" onClick={onLogout}>
            Logout
          </Button>
        ) : null}
      </CardFooter>
    </Card>
  );
}

export default Dashboard;
