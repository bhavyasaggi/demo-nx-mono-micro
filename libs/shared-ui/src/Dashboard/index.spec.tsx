import React from 'react';
import { render } from '@testing-library/react';

import Dashboard from './index';

describe('Dashboard', () => {
  it('should render successfully', () => {
    const { container } = render(
      <Dashboard
        id={15}
        email="kminchelle@qq.com"
        firstName="Jeanne"
        lastName="Halvorson"
        gender="female"
        image="https://robohash.org/autquiaut.png?size=50x50&set=set1"
      />
    );
    expect(container).toMatchSnapshot();
  });
});
