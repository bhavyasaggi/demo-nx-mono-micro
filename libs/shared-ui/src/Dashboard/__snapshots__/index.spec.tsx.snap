// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should render successfully 1`] = `
<div>
  <div
    class="chakra-card container css-2dlz7"
  >
    <div
      class="chakra-card__header css-0"
    >
      <h2
        class="chakra-heading css-0"
      >
        Welcome, 
        Jeanne
        !
      </h2>
    </div>
    <div
      class="chakra-card__body css-13a6xye"
    >
      <div
        class="chakra-table__container css-zipzvv"
      >
        <table
          class="chakra-table css-0"
        >
          <tbody
            class="css-0"
          >
            <tr
              class="css-0"
            >
              <td
                class="css-0"
              >
                <em
                  class="chakra-text css-0"
                >
                  Name
                </em>
              </td>
              <td
                class="css-0"
              >
                Jeanne Halvorson
              </td>
            </tr>
            <tr
              class="css-0"
            >
              <td
                class="css-0"
              >
                <em
                  class="chakra-text css-0"
                >
                  Gender
                </em>
              </td>
              <td
                class="css-0"
              >
                female
              </td>
            </tr>
            <tr
              class="css-0"
            >
              <td
                class="css-0"
              >
                <em
                  class="chakra-text css-0"
                >
                  Email
                </em>
              </td>
              <td
                class="css-0"
              >
                kminchelle@qq.com
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="css-gmuwbf"
      >
        <span
          class="chakra-avatar css-1uanvk6"
        >
          <div
            aria-label="Jeanne Halvorson"
            class="chakra-avatar__initials css-0"
            role="img"
          >
            JH
          </div>
        </span>
      </div>
    </div>
    <div
      class="chakra-card__footer css-k008qs"
    />
  </div>
</div>
`;
